core/utils/enums.py

```diff
@@ -0,0 +1,102 @@
+"""This file contains all the different enums used to implement callback
+pattern functionality."""
+
+from enum import Enum
+
+class JobTypes(Enum):
+    """
+    Enum to implement callback pattern for DMLRunner
+
+    JOB_TRAIN: Optimizer wants Runner to train the model
+
+    JOB_VAL: Optimizer wants Runner to validate the model
+
+    JOB_INIT: Optimizer wants Runner to initialize the model
+
+    JOB_AVG: Optimizer wants Runner to average weights via weighted avg.
+
+    JOB_COMM: Optimizer wants Runner to push the weights onto
+    IPFS and then communicate this to rest of nodes via blockchain
+    """
+    JOB_TRAIN = "TRAIN"
+    JOB_VAL = "VALIDATE"
+    JOB_INIT = "INITIALIZE"
+    JOB_AVG = "AVERAGE"
+    JOB_COMM = "COMMUNICATE"
+    NOTHING = "NOTHING"
```

Comment:
This should be removed -- there shouldn't exist a NOTHING job type.

Reply:
done & done